src/DotNetWorker.Core/IModelBindingData.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Azure.Functions.Worker.Core
+{
+    /// <summary>
+    /// A representation of a Microsoft.Azure.WebJobs.ParameterBindingData
+    /// </summary>
+    public interface IModelBindingData
```

Comment:
I still don't understand why this needs to be an interface. It seems unnecessary, and also means that we can't add things to the type in the future (e.g. new binding properties) without breaking contract changes.

Reply:
I think we have a thread about this here: https://github.com/Azure/azure-functions-dotnet-worker/pull/1107#discussion_r1010958793

I'm not opposed to removing the interface but want to double check with Fabio around his though process - I can sync with him offline and update this

Reply:
cc @fabiocav

Reply:
Resolving this one as we are discussing this under comment - https://github.com/Azure/azure-functions-dotnet-worker/pull/1107#discussion_r1002146117